src/mountscript.sh

```diff
@@ -602,6 +644,324 @@ ensure_aznfswatchdog()
+}
+
+
+#
+# Add stunnel configuration in stunnel_<storageaccount>.conf file.
+#
+add_stunnel_configuration()
+{
+    chattr -f -i $stunnel_conf_file
+
+    stunnel_CAFile="/etc/ssl/certs/DigiCert_Global_Root_G2.pem"
+
+    if [ ! -f $stunnel_CAFile ]; then
+        vecho "CA root cert is missing for stunnel configuration. Installing DigiCert_Global_Root_G2 certificate."
+        install_CA_cert
+        if [ $? -ne 0 ]; then
+            chattr -f +i $stunnel_conf_file
+            eecho "[FATAL] Not able to install DigiCert_Global_Root_G2 certificate!"
+            return 1
+        fi
+    fi
+
+    echo "CAFile = $stunnel_CAFile" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add CAFile path to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo "verifyChain = yes" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add verifyChain option to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo "sslVersion = TLSv1.2" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add sslVersion option to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo "debug = $DEBUG_LEVEL" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add debug option to $stunnel_conf_file!"
+        return 1
+    fi
+
+    stunnel_log_file="$STUNNELDIR/logs/stunnel_$nfs_host.log"
+    echo "output = $stunnel_log_file" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add log file path to $stunnel_conf_file!"
+        return 1
+    fi
+
+    stunnel_pid_file="$STUNNELDIR/logs/stunnel_$nfs_host.pid"
+    echo "pid = $stunnel_pid_file" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add pid file path to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo >> $stunnel_conf_file
+
+    echo "[$nfs_host]" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add $nfs_host service/entry name to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo "client = yes" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to 'client = yes' to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo "accept = $LOCALHOST:$available_port" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add 'accept' info to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo "connect = $nfs_host:$CONNECT_PORT" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add 'connect' info to $stunnel_conf_file!"
+        return 1
+    fi
+
```

Comment:
I put checkHost part on stash as my previous mount helper pr was messed up. And forgot to add stashed changes into this pr. But now added checkHost option in the stunnel configuration.
The checkHost value that I have used now works for devfabric and may work for preprod tenants. But for prod tenants, yes, we need to have a different checkHost value.
Instead of adding a flag, we can switch the checkHost value for preprod and prod tenants by checking $nfs_host value.
$nfs_host will have 'preprod' string for preprod tenants, so by parsing $nfs_host, we can check if its preprod or prod server and then switch the value of checkHost accordingly.